### PR TITLE
SW-5125: FE - Full Module View (Follow-ups)

### PR DESCRIPTION
**Pull request by joshbuchea**
This PR has some follow-ups including:

- Add module events to `ModuleService` & Redux
- Join Session buttons opens new browser window with event
- Add session call details box
- Update due date styles
- Connect module event links

## Tasks

- [ ] Update routes?
- [ ] Update `ModuleEvent` to include `module.name` & `module.dateRange` and remove module fetch?

### CODE REVIEW

**Comment by joshbuchea** [comment]
@tommylau523 @karthikbtf @nickgraz now that Nick is back, I think it would be good to get his opinion as well re: route changes. I'm happy to adjust the routes to whatever we'd prefer them to be. I don't have a preference for one convention vs another, so I'm happy to defer to the group.

@tommy I apologize for asking you to repeat yourself, but would you be willing share your proposed route changes again here in this PR? 🙏

**Comment by tommylau523** [comment]
Not a problem at all! 

currently we are following something like this:

`modules/project/:projectId`
`modules/project/:projectId/content/:moduleId`
`modules/project/:projectId/content/:moduleId/event/:eventId`
`modules/project/:projectId/content/:moduleId/resource/:contentId`

Which I think is totally fine as is. 

I had a suggestion that can reorganize routes to follow projects first:

`project/:projectId/module/:moduleId/event or content/:id`

It fits the resource hierarchy a bit better in my opinion.

**Comment by joshbuchea** [comment]
Awesome thanks @tommylau523!

**Comment by joshbuchea** [comment]
Also, right now the module needs to be fetched on the module events page because the module name & date range is used to render the page title. But if those properties were available in the module event data we could remove the module fetch. Any preference here?

File: src/scenes/ModulesRouter/ModuleEventView.tsx [hunk 1]
```diff
@@ -111,58 +69,85 @@ const ModuleEventView = () => {
...
+
+                {event?.additionalLinks?.length && (
+                  <>
+                    {event?.additionalLinks?.map((link, index) => (
+                      <Box key={index} marginBottom={theme.spacing(2)}>
+                        <Link
+                          fontSize='16px'
+                          onClick={() => {
+                            if (link.url) {
+                              window.open(link.url, '_blank', 'noopener noreferrer');
```
Comment (karthikbtf) [inline src/scenes/ModulesRouter/ModuleEventView.tsx:138]: Alternatively use the `to=` and `target=` properties of `Link`? Not sure if the `rel=` is auto added, maybe we could update Link to do that.

**Review by karthikbtf — APPROVED** [review]
Looks good to me, one suggestion.

File: src/scenes/ModulesRouter/ModuleEventView.tsx (hunk 1 above)
Comment (joshbuchea) [reply src/scenes/ModulesRouter/ModuleEventView.tsx:138]: So these URLs are external and it seemed like `Link` was specific for internal app routes, but I'll try it out and will update if `Link` handles external URLs.

**Comment by joshbuchea** [comment]
@nickgraz would love to get your thoughts when you have a free moment re: @tommylau523's comment about potentially updating the routes.

**Comment by nickgraz** [comment]
@joshbuchea I think those route suggestions are great. The ones I put in there were a complete shot in the dark, @tommylau523 has a better idea of how it will really be structured.

**Comment by joshbuchea** [comment]
Sounds good @nickgraz and @tommylau523, let's update the routes.

**Pull request merged**